ansible_collections/arista/avd/plugins/action/validate.py

```diff
@@ -0,0 +1,125 @@
+            raise AnsibleActionFail(f"Invalid value '{validation_mode}' for the argument 'validation_mode'."
+                                    f"Must be one of {VALID_VALIDATION_MODES}")
+
+        # ########
+        # READ ANSIBLE VARIABLES AND PERFORM TEMPLATING TO SUPPORT INLINE JINJA
+        hostname = task_vars['inventory_hostname']
+        data = {}
+        for var in task_vars:
+            if str(var).startswith(("ansible", "molecule", "hostvars", "vars")):
+                continue
+            try:
+                data[var] = self._templar.template(task_vars[var])
+            except Exception as e:
+                raise AnsibleActionFail(f"Exception during templating of task_var '{var}'") from e
+
+        try:
+            avd_schema = AvdSchema(schema)
+        except Exception as e:
+            raise AnsibleActionFail("Invalid Schema supplied to the arista.avd.validate plugin") from e
+
+        if conversion_mode != "disabled":
+            # auto-convert data according to schema (convert_types)
+            preconversion_data = copy.deepcopy(data)
+            exceptions = self.convert_data(data, avd_schema)
+            conversion_counter = self.handle_exceptions(exceptions, conversion_mode, hostname)
+            if data != preconversion_data:
+                # Idempotency checks fail when we set the changed flag on auto upgrade.
+                # Uncomment this when all data models and molecule tests has been updated.
+                # result['changed'] = True
+                result.setdefault('ansible_facts', {})
+                for key, value in data.items():
+                    if key not in preconversion_data or value != preconversion_data[key]:
+                        result['ansible_facts'][key] = value
+            if conversion_counter:
+                result['msg'] = f"{conversion_counter} conversions done. Check converted facts with -v or -vvv."
+
+        if validation_mode != "disabled":
+            # perform schema validation according to schema
+            exceptions = self.validate_data(data, avd_schema)
+            validation_counter = self.handle_exceptions(exceptions, validation_mode, hostname)
+            if validation_counter and validation_mode == "error":
+                result['ansible_facts'] = {}
+                result['changed'] = False
+                result['failed'] = True
+            if result.get('msg'):
+                result['msg'] += " "
+            if validation_counter:
+                result['msg'] = f"{result.get('msg', '')}{validation_counter} errors found during schema validation of input vars."
+
+        return result
+
+    def convert_data(self, data, avd_schema):
+        output = []
+        exceptions = avd_schema.convert(data)
+        # avd_schema.convert returns a generator, which we need to run over to perform the actual conversions.
+        for exception in exceptions:
+            output.append(exception)
+        return output
```

Comment:
Would this not work?
```suggestion
        exceptions = avd_schema.convert(data)
        # avd_schema.convert returns a generator, which we need to run over to perform the actual conversions.
        return list(exceptions)
```

could even be simplified further if it works:

```python
return list(avd_schema.convert(data))
```

Reply:
fixed